File: Android/tests/src/jp/co/riso/android/util/LoggerTest.java
package jp.co.riso.android.util;

import jp.co.riso.smartdeviceapp.view.MainActivity;
import android.test.ActivityInstrumentationTestCase2;
import android.test.UiThreadTest;

public class LoggerTest extends ActivityInstrumentationTestCase2<MainActivity> {
    
    public LoggerTest() {
        super(MainActivity.class);
    }
    
    public LoggerTest(Class<MainActivity> activityClass) {
        super(activityClass);
    }

    protected void setUp() throws Exception {
        super.setUp();
    }

    protected void tearDown() throws Exception {
        super.tearDown();
    }
    
    //================================================================================
    // Test Logs
    //================================================================================
    
    public void testLogs() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, false, false);
        try {
            Logger.logVerbose(LoggerTest.class, "Log Message");
            Logger.logWarn(LoggerTest.class, "Log Message");
            Logger.logInfo(LoggerTest.class, "Log Message");
            Logger.logDebug(LoggerTest.class, "Log Message");
            Logger.logError(LoggerTest.class, "Log Message");
            Logger.logAssert(LoggerTest.class, "Log Message");
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testLogs_Off() {
        Logger.initialize(Logger.LOGLEVEL_NONE, false, false);
        try {
            Logger.logVerbose(LoggerTest.class, "Log Message");
            Logger.logWarn(LoggerTest.class, "Log Message");
            Logger.logInfo(LoggerTest.class, "Log Message");
            Logger.logDebug(LoggerTest.class, "Log Message");
            Logger.logError(LoggerTest.class, "Log Message");
            Logger.logAssert(LoggerTest.class, "Log Message");
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testLogs_NullClass() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, false, false);
        try {
            Logger.logVerbose(null, "Log Message");
            Logger.logWarn(null, "Log Message");
            Logger.logInfo(null, "Log Message");
            Logger.logDebug(null, "Log Message");
            Logger.logError(null, "Log Message");
            Logger.logAssert(null, "Log Message");
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testLogs_WithParameters() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, false, false);
        try {
            Logger.logVerbose(LoggerTest.class, "Log Message %d", 123);
            Logger.logWarn(LoggerTest.class, "Log Message %d", 123);
            Logger.logInfo(LoggerTest.class, "Log Message %d", 123);
            Logger.logDebug(LoggerTest.class, "Log Message %d", 123);
            Logger.logError(LoggerTest.class, "Log Message %d", 123);
            Logger.logAssert(LoggerTest.class, "Log Message %d", 123);
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testLogs_WithInvalidParameters() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, false, false);
        try {
            Logger.logVerbose(LoggerTest.class, "Log Message %d", "asd");
            Logger.logWarn(LoggerTest.class, "Log Message %d", "asd");
            Logger.logInfo(LoggerTest.class, "Log Message %d", "asd");
            Logger.logDebug(LoggerTest.class, "Log Message %d", "asd");
            Logger.logError(LoggerTest.class, "Log Message %d", "asd");
            Logger.logAssert(LoggerTest.class, "Log Message %d", "asd");
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testLogs_WithParamInvalidFormat() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, false, false);
        try {
            Logger.logVerbose(LoggerTest.class, "Log Message", "asd");
            Logger.logWarn(LoggerTest.class, "Log Message", "asd");
            Logger.logInfo(LoggerTest.class, "Log Message", "asd");
            Logger.logDebug(LoggerTest.class, "Log Message", "asd");
            Logger.logError(LoggerTest.class, "Log Message", "asd");
            Logger.logAssert(LoggerTest.class, "Log Message", "asd");
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testLogs_WithIncompleteFormat() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, false, false);
        try {
            Logger.logVerbose(LoggerTest.class, "Log Message %d");
            Logger.logWarn(LoggerTest.class, "Log Message %d");
            Logger.logInfo(LoggerTest.class, "Log Message %d");
            Logger.logDebug(LoggerTest.class, "Log Message %d");
            Logger.logError(LoggerTest.class, "Log Message %d");
            Logger.logAssert(LoggerTest.class, "Log Message %d");
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testLogs_WithMoreParam() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, false, false);
        try {
            Logger.logVerbose(LoggerTest.class, "Log Message %d", 123, "asd");
            Logger.logWarn(LoggerTest.class, "Log Message %d", 123, "asd");
            Logger.logInfo(LoggerTest.class, "Log Message %d", 123, "asd");
            Logger.logDebug(LoggerTest.class, "Log Message %d", 123, "asd");
            Logger.logError(LoggerTest.class, "Log Message %d", 123, "asd");
            Logger.logAssert(LoggerTest.class, "Log Message %d", 123, "asd");
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testLogs_NullFormat() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, false, false);
        try {
            Logger.logVerbose(LoggerTest.class, null, 123, "asd");
            Logger.logWarn(LoggerTest.class, null, 123, "asd");
            Logger.logInfo(LoggerTest.class, null, 123, "asd");
            Logger.logDebug(LoggerTest.class, null, 123, "asd");
            Logger.logError(LoggerTest.class, null, 123, "asd");
            Logger.logAssert(LoggerTest.class, null, 123, "asd");
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    //================================================================================
    // Test Get Log
    //================================================================================
    
    //================================================================================
    // Test Start Stop Time
    //================================================================================
    
    public void testStartStopTime() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, false);

        try {
            Logger.logStartTime(getActivity(), LoggerTest.class, "PROCESS1");
            Thread.sleep(500);
            Logger.logStopTime(getActivity(), LoggerTest.class, "PROCESS1");
        } catch (InterruptedException e) {
            
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testStartStopTime_Off() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, false, false);

        try {
            Logger.logStartTime(getActivity(), LoggerTest.class, "PROCESS1");
            Thread.sleep(500);
            Logger.logStopTime(getActivity(), LoggerTest.class, "PROCESS1");
        } catch (InterruptedException e) {
            
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testStartStopTime_invalidContext() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, false);

        try {
            Logger.logStartTime(null, LoggerTest.class, "PROCESS1");
            Thread.sleep(500);
            Logger.logStopTime(null, LoggerTest.class, "PROCESS1");
        } catch (InterruptedException e) {
            
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testStartStopTime_InvalidClass() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, false);

        try {
            Logger.logStartTime(getActivity(), null, "PROCESS1");
            Thread.sleep(500);
            Logger.logStopTime(getActivity(), null, "PROCESS1");
        } catch (InterruptedException e) {
            
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testStartStopTime_InvalidProcess() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, false);

        try {
            Logger.logStartTime(getActivity(), LoggerTest.class, null);
            Thread.sleep(500);
            Logger.logStopTime(getActivity(), LoggerTest.class, null);
        } catch (InterruptedException e) {
            
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testStartStopTime_EmptyProcess() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, false);

        try {
            Logger.logStartTime(getActivity(), LoggerTest.class, "");
            Thread.sleep(500);
            
            Logger.logStopTime(getActivity(), LoggerTest.class, "");
        } catch (InterruptedException e) {
            
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testStartStopTime_StartTwice() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, true);

        try {
            Logger.logStartTime(getActivity(), LoggerTest.class, "PROCESS1");
            Thread.sleep(500);
            Logger.logStartTime(getActivity(), LoggerTest.class, "PROCESS1");
            Thread.sleep(500);
            Logger.logStopTime(getActivity(), LoggerTest.class, "PROCESS1");
            
            assertTrue(Logger.getLogString(getActivity()) != null);
        } catch (InterruptedException e) {
            
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testStartStopTime_StopWithoutStart() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, true);

        try {
            Logger.logStopTime(getActivity(), LoggerTest.class, "PROCESS1");
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    public void testStartStopTime_WriteToFile() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, true);

        try {
            Logger.logStartTime(getActivity(), LoggerTest.class, "PROCESS1");
            Thread.sleep(1000);
            Logger.logStopTime(getActivity(), LoggerTest.class, "PROCESS1");
            
            assertTrue(Logger.getLogString(getActivity()) != null);
        } catch (InterruptedException e) {
            
        } catch (Exception e) {
            fail("Should not throw any exception");
        }
    }
    
    @UiThreadTest
    public void testGetLogString() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, true);
        Logger.logStartTime(getActivity(), LoggerTest.class, "PROCESS1");
        assertTrue(Logger.getLogString(getActivity()) != null);
    }

    @UiThreadTest
    public void testGetLogString_NullContext() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, true);
        Logger.logStartTime(getActivity(), LoggerTest.class, "PROCESS1");
        assertTrue(Logger.getLogString(null) == null);
    }

    @UiThreadTest
    public void testWriteToFile() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, true);
        Logger.writeToFile(getActivity(), "Hello");
        assertTrue(Logger.getLogString(getActivity()) != null);
    }

    @UiThreadTest
    public void testWriteToFile_NullContext() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, true);
        Logger.writeToFile(null, "Hello");
        assertTrue(Logger.getLogString(getActivity()) == null);
    }

    @UiThreadTest
    public void testWriteToFile_NullMessage() {
        Logger.initialize(Logger.LOGLEVEL_VERBOSE, true, true);
        Logger.writeToFile(getActivity(), null);
        assertTrue(Logger.getLogString(getActivity()) != null);
    }
    
    public void testDeleteTask() {
        Logger.runDeleteTask(getActivity());
    }
    
    public void testDeleteTask_NullContext() {
        Logger.runDeleteTask(null);
    }
}

File: Android/app/src/androidTest/java/jp/co/riso/android/util/LoggerTest.kt
package jp.co.riso.android.util

import jp.co.riso.android.util.Logger.getLogString
import jp.co.riso.android.util.Logger.initialize
import jp.co.riso.android.util.Logger.logAssert
import jp.co.riso.android.util.Logger.logDebug
import jp.co.riso.android.util.Logger.logError
import jp.co.riso.android.util.Logger.logInfo
import jp.co.riso.android.util.Logger.logStartTime
import jp.co.riso.android.util.Logger.logStopTime
import jp.co.riso.android.util.Logger.logVerbose
import jp.co.riso.android.util.Logger.logWarn
import jp.co.riso.android.util.Logger.runDeleteTask
import jp.co.riso.android.util.Logger.writeToFile
import jp.co.riso.smartdeviceapp.view.BaseActivityTestUtil
import junit.framework.TestCase
import org.junit.Test

class LoggerTest : BaseActivityTestUtil() {

    //================================================================================
    // Test Logs
    //================================================================================
    @Test
    fun testLogs() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = false, perfLogsToFile = false)
        try {
            logVerbose(LoggerTest::class.java, "Log Message")
            logWarn(LoggerTest::class.java, "Log Message")
            logInfo(LoggerTest::class.java, "Log Message")
            logDebug(LoggerTest::class.java, "Log Message")
            logError(LoggerTest::class.java, "Log Message")
            logAssert(LoggerTest::class.java, "Log Message")
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testLogs_Off() {
        initialize(Logger.LOGLEVEL_NONE, perfLogs = false, perfLogsToFile = false)
        try {
            logVerbose(LoggerTest::class.java, "Log Message")
            logWarn(LoggerTest::class.java, "Log Message")
            logInfo(LoggerTest::class.java, "Log Message")
            logDebug(LoggerTest::class.java, "Log Message")
            logError(LoggerTest::class.java, "Log Message")
            logAssert(LoggerTest::class.java, "Log Message")
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testLogs_NullClass() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = false, perfLogsToFile = false)
        try {
            logVerbose(null, "Log Message")
            logWarn(null, "Log Message")
            logInfo(null, "Log Message")
            logDebug(null, "Log Message")
            logError(null, "Log Message")
            logAssert(null, "Log Message")
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testLogs_WithParameters() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = false, perfLogsToFile = false)
        try {
            logVerbose(LoggerTest::class.java, "Log Message %d", 123)
            logWarn(LoggerTest::class.java, "Log Message %d", 123)
            logInfo(LoggerTest::class.java, "Log Message %d", 123)
            logDebug(LoggerTest::class.java, "Log Message %d", 123)
            logError(LoggerTest::class.java, "Log Message %d", 123)
            logAssert(LoggerTest::class.java, "Log Message %d", 123)
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testLogs_WithInvalidParameters() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = false, perfLogsToFile = false)
        try {
            logVerbose(LoggerTest::class.java, "Log Message %d", "asd")
            logWarn(LoggerTest::class.java, "Log Message %d", "asd")
            logInfo(LoggerTest::class.java, "Log Message %d", "asd")
            logDebug(LoggerTest::class.java, "Log Message %d", "asd")
            logError(LoggerTest::class.java, "Log Message %d", "asd")
            logAssert(LoggerTest::class.java, "Log Message %d", "asd")
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testLogs_WithParamInvalidFormat() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = false, perfLogsToFile = false)
        try {
            logVerbose(LoggerTest::class.java, "Log Message", "asd")
            logWarn(LoggerTest::class.java, "Log Message", "asd")
            logInfo(LoggerTest::class.java, "Log Message", "asd")
            logDebug(LoggerTest::class.java, "Log Message", "asd")
            logError(LoggerTest::class.java, "Log Message", "asd")
            logAssert(LoggerTest::class.java, "Log Message", "asd")
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testLogs_WithIncompleteFormat() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = false, perfLogsToFile = false)
        try {
            logVerbose(LoggerTest::class.java, "Log Message %d")
            logWarn(LoggerTest::class.java, "Log Message %d")
            logInfo(LoggerTest::class.java, "Log Message %d")
            logDebug(LoggerTest::class.java, "Log Message %d")
            logError(LoggerTest::class.java, "Log Message %d")
            logAssert(LoggerTest::class.java, "Log Message %d")
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testLogs_WithMoreParam() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = false, perfLogsToFile = false)
        try {
            logVerbose(LoggerTest::class.java, "Log Message %d", 123, "asd")
            logWarn(LoggerTest::class.java, "Log Message %d", 123, "asd")
            logInfo(LoggerTest::class.java, "Log Message %d", 123, "asd")
            logDebug(LoggerTest::class.java, "Log Message %d", 123, "asd")
            logError(LoggerTest::class.java, "Log Message %d", 123, "asd")
            logAssert(LoggerTest::class.java, "Log Message %d", 123, "asd")
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testLogs_NullFormat() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = false, perfLogsToFile = false)
        try {
            logVerbose(LoggerTest::class.java, null, 123, "asd")
            logWarn(LoggerTest::class.java, null, 123, "asd")
            logInfo(LoggerTest::class.java, null, 123, "asd")
            logDebug(LoggerTest::class.java, null, 123, "asd")
            logError(LoggerTest::class.java, null, 123, "asd")
            logAssert(LoggerTest::class.java, null, 123, "asd")
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    //================================================================================
    // Test Get Log
    //================================================================================
    //================================================================================
    // Test Start Stop Time
    //================================================================================
    @Test
    fun testStartStopTime() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = false)
        try {
            logStartTime(mainActivity, LoggerTest::class.java, "PROCESS1")
            Thread.sleep(500)
            logStopTime(mainActivity, LoggerTest::class.java, "PROCESS1")
        } catch (e: InterruptedException) {
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testStartStopTime_Off() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = false, perfLogsToFile = false)
        try {
            logStartTime(mainActivity, LoggerTest::class.java, "PROCESS1")
            Thread.sleep(500)
            logStopTime(mainActivity, LoggerTest::class.java, "PROCESS1")
        } catch (e: InterruptedException) {
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testStartStopTime_invalidContext() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = false)
        try {
            logStartTime(null, LoggerTest::class.java, "PROCESS1")
            Thread.sleep(500)
            logStopTime(null, LoggerTest::class.java, "PROCESS1")
        } catch (e: InterruptedException) {
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testStartStopTime_InvalidClass() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = false)
        try {
            logStartTime(mainActivity, null, "PROCESS1")
            Thread.sleep(500)
            logStopTime(mainActivity, null, "PROCESS1")
        } catch (e: InterruptedException) {
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testStartStopTime_InvalidProcess() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = false)
        try {
            logStartTime(mainActivity, LoggerTest::class.java, null)
            Thread.sleep(500)
            logStopTime(mainActivity, LoggerTest::class.java, null)
        } catch (e: InterruptedException) {
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testStartStopTime_EmptyProcess() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = false)
        try {
            logStartTime(mainActivity, LoggerTest::class.java, "")
            Thread.sleep(500)
            logStopTime(mainActivity, LoggerTest::class.java, "")
        } catch (e: InterruptedException) {
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testStartStopTime_StartTwice() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = true)
        try {
            logStartTime(mainActivity, LoggerTest::class.java, "PROCESS1")
            Thread.sleep(500)
            logStartTime(mainActivity, LoggerTest::class.java, "PROCESS1")
            Thread.sleep(500)
            logStopTime(mainActivity, LoggerTest::class.java, "PROCESS1")
            TestCase.assertTrue(getLogString(mainActivity) != null)
        } catch (e: InterruptedException) {
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testStartStopTime_StopWithoutStart() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = true)
        try {
            logStopTime(mainActivity, LoggerTest::class.java, "PROCESS1")
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testStartStopTime_WriteToFile() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = true)
        try {
            logStartTime(mainActivity, LoggerTest::class.java, "PROCESS1")
            Thread.sleep(1000)
            logStopTime(mainActivity, LoggerTest::class.java, "PROCESS1")
            TestCase.assertTrue(getLogString(mainActivity) != null)
        } catch (e: InterruptedException) {
        } catch (e: Exception) {
            TestCase.fail("Should not throw any exception")
        }
    }

    @Test
    fun testGetLogString() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = true)
        logStartTime(mainActivity, LoggerTest::class.java, "PROCESS1")
        TestCase.assertTrue(getLogString(mainActivity) != null)
    }

    @Test
    fun testGetLogString_NullContext() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = true)
        logStartTime(mainActivity, LoggerTest::class.java, "PROCESS1")
        TestCase.assertTrue(getLogString(null) == null)
    }

    @Test
    fun testWriteToFile() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = true)
        writeToFile(mainActivity, "Hello")
        TestCase.assertTrue(getLogString(mainActivity) != null)
    }

    @Test
    fun testWriteToFile_NullContext() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = true)
        writeToFile(null, "Hello")
        TestCase.assertTrue(getLogString(mainActivity) == null)
    }

    @Test
    fun testWriteToFile_NullMessage() {
        initialize(Logger.LOGLEVEL_VERBOSE, perfLogs = true, perfLogsToFile = true)
        writeToFile(mainActivity, null)
        TestCase.assertTrue(getLogString(mainActivity) != null)
    }

    @Test
    fun testDeleteTask() {
        runDeleteTask(mainActivity)
    }

    @Test
    fun testDeleteTask_NullContext() {
        runDeleteTask(null)
    }
}